agent/command/archive_tarball_create.go

```diff
@@ -96,7 +96,13 @@ func (c *tarballCreate) Execute(ctx context.Context,
 		}()
```

Comment:
Won't let me comment on unchanged lines, but L94 needs to exit if the context errors as well, because the defer will hang if there's no consumer for `errChan`.